interface/src/app/gene/network/network.js

```diff
@@ -0,0 +1,262 @@
+/**
+ * "adage.gene.network" module.
+ */
+
+angular.module('adage.gene.network', [
+  'ui.router',
+  'placeholders',
+  'ui.bootstrap',
+  'ui.slider'
+])
+
+.config(function config($stateProvider) {
+  $stateProvider.state('gene_network', {
+    url: '/gene_network/?genes&mlmodel',
+    views: {
+      main: {
+        templateUrl: 'gene/network/network.tpl.html',
+        controller: 'GeneNetworkCtrl',
+        controllerAs: 'ctrl'
+      }
+    },
+    data: {pageTitle: 'Gene Network'}
+  });
+})
+
+.factory('EdgeService', ['$resource', function($resource) {
+  return $resource('/api/v0/edge/');
+}])
+
+.factory('NodeService', ['$resource', function($resource) {
+  return $resource('/api/v0/node/');
+}])
+
+.controller('GeneNetworkCtrl',
+   ['$stateParams', 'EdgeService', 'NodeService', '$log',
+    function GeneNetworkController($stateParams, EdgeService, NodeService,
+                                   $log) {
+      var rawMinWeight = -1.0;
+      var rawMaxWeight = 1.0;
+      var scaledMaxWeight = rawMaxWeight - rawMinWeight;
+      var self = this;
+      // The following properties of "self" will be available to HTMl.
+      self.minCorrelation = -1.0;
+      self.maxNodeNum = Number.MAX_SAFE_INTEGER;
+      self.statusMessage = 'Connecting to the server ...';
+      self.sliderOptions = {  // Slider configuration.
+        range: 'max',
+        stop: function(event, ui) {
+          network.filter(self.minCorrelation - rawMinWeight,
+                         self.maxNodeNum).draw();
+        }
+      };
+
+      // Do nothing if no genes are specified in URL.
+      if (!$stateParams.genes || !$stateParams.genes.split(',').length) {
+        self.statusMessage = 'No genes are specified.';
+        return;
+      }
+
+      var urlGenes = $stateParams.genes.split(',');
+      var setEdgeColor = d3.scale.linear()
+          .domain([0, scaledMaxWeight / 2.0, scaledMaxWeight])
+          .range(['green', 'orange', 'red']);
+
+      var genes = [];
+      var edges = [];
+      var network = d3.network()  // Initialize the network.
```

Comment:
So I guess all of this code is using d3 and not vega at all for the visualization right? Also I guess we don't need to include d3 as one of the dependencies for this module at the top of this file since it is not an angular dependency?